public/src/shared/header.js

```diff
@@ -1,11 +1,193 @@
+import {useState} from "react"
 import {Link} from "react-router-dom";
+import {AiOutlineMenu, BsBag, IoPersonCircle, GrClose} from "react-icons/all";
 
 function Header() {
+    const [isShown, setIsShown] = useState({})
+    const [sideMenu, setSideMenu] = useState(false)
+    const [dropdownOpen, setDropdownOpen] = useState(false)
+
+    const links = [
```

Comment:
I would move this outside of the function. Not much time in declaration but it helps not to be declared every time the component changes.

Reply:
Smart. Didn't even think about it as it'll be replaced when we get a backend to link to. I'll fix it!